Clean up digest emails.

showell opened the pull request.

timabbott commented:
I merged all but the last commit as the series ending with 88a57ed4ac5631b6664dda02d1c117690aaa23ad.

I don't understand the time logic for the tests in the last commit -- I think it's possible for the `one_minute_ago` logic to fail if we (re)built the test database in the minute before running the test, which might actually happen in CI.

showell commented:
I updated the data-related test as per our PM conversation (and rebased).

Ready for review again.  (Also, more changes will be coming in the next day or two--including some more substantive performance improvements.)

timabbott commented:
Merged, thanks @showell!
